### PR TITLE
First pass at resource copy

### PR DESCRIPTION
**Pull request by aphelionz**

### CODE REVIEW

File: hs_core/tests/api/rest/test_copy_resource.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+class TestPublicCopyResourceEndpoint(HSRESTTestCase):
+    def setUp(self):
+        super(TestPublicCopyResourceEndpoint, self).setUp()
+
+        self.rtype = 'GenericResource'
+        self.title = 'My Test resource'
+        res = resource.create_resource(self.rtype,
+                                       self.user,
+                                       self.title,
+                                       unpack_file=False)
```
Comment (hyi) [inline hs_core/tests/api/rest/test_copy_resource.py]: unpack_file=False is the default and not needed here.

File: hs_core/tests/api/rest/test_copy_resource.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+        self.pid = res.short_id
+        self.resources_to_delete.append(self.pid)
+
+    def test_copy(self):
+        copy_url = "/hsapi/resource/%s/copy/" % self.pid
+        response = self.client.post(copy_url, {}, format='json')
+        new_pid = response.url.split('/')[4]
+        self.resources_to_delete.append(new_pid)
```
Comment (hyi) [inline hs_core/tests/api/rest/test_copy_resource.py]: Just curious where this self.resources_to_delete list is used to actually delete all created resources. I don't see the resource deletion part from code changes.

File: hs_core/urls.py [hunk 3]
```diff
@@ -15,13 +15,13 @@
     url(r'^resourceList/$', views.resource_rest_api.ResourceList.as_view(),
         name='DEPRECATED_list_resources'),
 
-    url(r'^resource/$', views.resource_rest_api.ResourceListCreate.as_view(),
-        name='list_create_resource'),
-
```
Comment (hyi) [inline hs_core/urls.py]: Why does this resource create REST endpoint be deleted? It appears to me the REST api class in resource_rest_api.py is still in our codebase and used for resource creation or listing.

File: hs_core/tests/api/rest/test_copy_resource.py (hunk 2 above)
Comment (aphelionz) [inline hs_core/tests/api/rest/test_copy_resource.py]: It happens in the tearDown function of the inherited HSRestTestCase class

File: hs_core/tests/api/rest/test_copy_resource.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+        self.rtype = 'GenericResource'
+        self.title = 'My Test resource'
+        res = resource.create_resource(self.rtype,
+                                       self.user,
+                                       self.title)
+
+        self.pid = res.short_id
+        self.resources_to_delete.append(self.pid)
+
+    def test_copy(self):
```
Comment (pkdash) [inline hs_core/tests/api/rest/test_copy_resource.py]: ```test_copy()``` -> ``test_copy_resource()```

File: hs_core/views/__init__.py [hunk 5]
```diff
@@ -540,6 +541,11 @@ def copy_resource(request, shortkey, *args, **kwargs):
     return HttpResponseRedirect(new_resource.get_absolute_url())
 
 
+@api_view(['POST'])
+def copy_resource_public(request, pk):
```
Comment (pkdash) [inline hs_core/views/__init__.py]: Can we make this function to return id of the copied resource and not the resource url?

**Comment by pkdash** [comment]
@aphelionz Needs resolving conflicts.

File: hs_core/views/__init__.py (hunk 5 above)
Comment (aphelionz) [inline hs_core/views/__init__.py]: Should be fixed

**Comment by aphelionz** [comment]
@pkdash Fixed the conflicts. Please let me know if you can +!

**Comment by pkdash** [comment]
@aphelionz +1. Merge after Jenkins checks are successful.  Thanks.

**Comment by pkdash** [comment]
@hyi See if you can +1.

**Pull request merged**